In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv("../data/raw/transactions.csv")

print("Rows:", len(df))
print("Columns:", list(df.columns))
print()
df.head(10)


Rows: 2000
Columns: ['transaction_id', 'date', 'description', 'amount', 'true_category']



,transaction_id,date,description,amount,true_category
0,T01525,2023-12-05,CIRCLE K # 99914,39.69,Gas
1,T00848,2023-11-13,DOORDASH*MCDONALDS,45.51,Restaurants
2,T01587,2023-11-07,UBER* EATS HELP.UBER.COM,3.29,Transport
3,T00597,2023-01-18,RITE AID #5898 DENVER CO,10.05,Health
4,T01071,2023-01-18,SQ *BLUE BOTTLE 9722 DALLAS TX,6.64,Coffee
5,T00693,2023-11-02,SQ *THE COMEDY CLUB CHICAGO IL,8.97,Entertainment
6,T00659,2023-06-07,DART RAIL DALLAS TX,56.47,Transport
7,T00525,2023-12-25,UBER* EATS HELP.UBER.COM,30.14,Transport
8,T00628,2023-07-03,DUNKIN #9922 PHX AZ,9.19,Coffee
9,T01622,2023-03-31,steamgames.com 465 wa,59.52,Entertainment


In [2]:
print("Category counts:")
print(df["true_category"].value_counts())
print()
print("Number of categories:", df["true_category"].nunique())

Category counts:
true_category
Groceries        270
Utilities        246
Transport        235
Entertainment    235
Gas              218
Health           210
Coffee           210
Shopping         192
Restaurants      184
Name: count, dtype: int64

Number of categories: 9


In [3]:
print("Sample descriptions (the text our classifier must interpret):")
for desc in df["description"].sample(15, random_state=1):
    print("  ", repr(desc))

Sample descriptions (the text our classifier must interpret):
   'oncor electric delivery'
   'ONCOR ELECTRIC DELIVERY'
   'KROGER 64651 AUSTIN TX'
   'ETSY.COM - SELLER84'
   'NETFLIX.COM 760'
   'TST* OLIVE & VINE AUSTIN TX'
   'POS DEBIT QUEST DIAGNOSTIC'
   'ALDI  592 PHX AZ'
   "TRADER JOE'S #48"
   'TST* OLIVE & VINE CHICAGO IL'
   'SQ *CITY DENTAL CHICAGO IL'
   'SPOTIFY USA 3'
   'SPOTIFY USA 49  '
   'WALGREENS #50307'
   'CIRCLE K # 81143'


In [4]:
print("Missing values per column:")
print(df.isnull().sum())
print()
print("Duplicate rows:", df.duplicated().sum())
print()
print("Amount range: $", df["amount"].min(), "to $", df["amount"].max())

Missing values per column:
transaction_id    0
date              0
description       0
amount            0
true_category     0
dtype: int64

Duplicate rows: 0

Amount range: $ 2.22 to $ 478.04


## Day 1 — Data Exploration Notes

- 2,000 transactions across 9 spend categories, roughly balanced (184–270 each).
- Descriptions are messy in realistic ways: inconsistent casing, doubled/trailing
  spaces, payment-processor prefixes (TST*, SQ*, POS DEBIT), and store numbers.
- No missing values or duplicates — data quality is clean, so the work is text
  classification, not data repair.
- Built-in ambiguity to watch: "UBER* EATS" is labeled Transport but is really
  food delivery — a known hard case for the evaluation.